### DISCUSSION TITLE
Routes with multiple unknown URL segments

### DISCUSSION THREAD

**Discussion by darylknight** [craftcms/cms#14341] [Ideas] [answered]
We recently implemented a Commerce7 integration for a client to a Craft site. The Commerce7 [docs](https://design-docs.commerce7.com/docs/home#3-create-routes-inside-your-cms) tell you to set up routes in the CMS such as `profile/*`, but it's become clear over the last few weeks that their documentation is incomplete when it comes to how many URL segments there will be for "*".

Initially I set up a rote for `profile/*`, but they actually generate all kind of URLS that _start_ with `profile/`, but so far have up to 4 further URL segments after that. Craft shows 404 errors for anything with more than one URL segment after `profile/`. I've set up the following routes to cover that, but I'm wondering if there's a way for "*" to be a literal wildcard for the number of URL segments too, rather than just "something in exactly one URL segment"?

![image](https://github.com/craftcms/cms/assets/1250017/9bdb128b-58fe-4079-94ff-5c3ed87519ef)

**Comment by brandonkelly** [comment, marked as answer]
Anything beyond the basics is what [URL rules](https://craftcms.com/docs/4.x/routing.html#advanced-routing-with-url-rules) are for (`config/routes.php`).

In your case that would be:

```php
return [
    'profile/<subpath:.+>' => ['template' => '_commerce7/profile'],
];
